parsl/tests/test_staging/test_file.py

```diff
@@ -22,11 +22,11 @@ def test_files():
 
 
 @pytest.mark.local
-def test_open():
-    with open('test-open.txt', 'w') as tfile:
-        tfile.write('Hello')
+def test_open(tmpd_cwd):
+    fpath = tmpd_cwd / 'test-open.txt'
+    fpath.write_text('Hello')
 
-    pfile = File('test-open.txt')
+    pfile = File(fpath)
 
-    with open(str(pfile), 'r') as opfile:
-        assert (opfile.readlines()[0] == 'Hello')
+    with open(pfile) as opfile:
```

Comment:
this is changing what's being tested: before, it tested that whatever a File returned as its `str` could be opened. Now it's testing that whatever comes back as its `fspath` can be opened.

That `str` implementation is a bit horrible (it raises an exception for File objects which don't have a local representation) but I think it's still needed as part of usability: for example, people writing bash_apps would traditionally write `f"cat {myfile}"` when forming their bash command lines rather than `f"cat {os.fspath(myfile)}"

So maybe this should be testing *both* behaviours?